src/components/timeline/TrackMemoryGraph.js

```diff
@@ -84,7 +84,15 @@ class TrackMemoryCanvas extends React.PureComponent<CanvasProps> {
     canvas.height = Math.round(deviceHeight);
     ctx.clearRect(0, 0, deviceWidth, deviceHeight);
 
-    const samples = counter.sampleGroups.samples;
+    const sampleGroups = counter.sampleGroups;
+    if (sampleGroups.length !== 1) {
```

Comment:
We are checking this here but it's not possible to fail here actually, checking here just in case. Because we only send one element from gecko, and if we fail to send that element(send an empty array instad), we handle this before coming here and remove that counter all together.

Reply:
I'd like to challenge this assumption: this is the case now, but will it be the case in the future? For example with fission in the future, we could have different sampleGroups for one process. (we don't know :) ).

I'm not saying we should account for this now. But I think that the condition should be `sampleGroups.length === 0` instead, and that we should throw because if that shouldn't happen we shouldn't silently ignore the problem.

Reply:
> Throwing an error seems too much to me. If Firefox didn't capture any counter samples, that shouldn't make the whole front-end crash. 

That's not what I suggest :)
I suggest that we throw _here_ because that shouldn't happen here. Indeed we should never add a TrackMemoryGraph if Firefox didn't capture any counter samples, because we should have filtered them out earlier.
That's why I sugges to throw here: because that should never happen. As you said `it's not possible to fail here actually` :)